src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RazorServiceDescriptors.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor
+{
+    internal static class RazorServiceDescriptors
+    {
+        private const string ComponentName = "Razor";
+
+        private static readonly ImmutableArray<JsonConverter> JsonConverters = new JsonConverterCollection()
+            .RegisterRazorConverters()
+            .ToImmutableArray();
+
+        public static readonly RazorServiceDescriptorsWrapper TagHelperProviderServiceDescriptors = new RazorServiceDescriptorsWrapper(ComponentName, _ => "Razor TagHelper Provider", JsonConverters, new (Type, Type?)[] { (typeof(IRemoteTagHelperProviderService), null) });
```

Comment:
@tmat I have this static right now with the assumption that during the lifetime of the process the service descriptor should be the same and shouldn't change under the covers. Hopefully that's an "OK" assumption.

Reply:
Yes, we use static descriptors as well.